src/drive/storage/index.ts

```diff
@@ -0,0 +1,244 @@
+  private readonly axios: AxiosStatic;
+  private readonly apiUrl: string;
+  private readonly clientName: string;
+  private readonly clientVersion: string;
+  private readonly token: Token;
+  private readonly mnemonic: string;
+
+  public static client(
+    apiUrl: string, clientName: string, clientVersion: string, token: Token, mnemonic: string
+  ) {
+    return new Storage(axios, apiUrl, clientName, clientVersion, token, mnemonic);
+  }
+
+  constructor(
+    axios: AxiosStatic, apiUrl: string, clientName: string, clientVersion: string, token: Token, mnemonic: string
+  ) {
+    this.axios = axios;
+    this.apiUrl = apiUrl;
+    this.clientName = clientName;
+    this.clientVersion = clientVersion;
+    this.token = token;
+    this.mnemonic = mnemonic;
+  }
+
+  /**
+   * Creates a new folder
+   * @param payload
+   */
+  public createFolder(payload: CreateFolderPayload): [
+    Promise<CreateFolderResponse>,
+    CancelTokenSource
+  ] {
+    const cancelTokenSource = axios.CancelToken.source();
+    const promise = this.axios
+      .post(`${this.apiUrl}/api/storage/folder`, {
+        parentFolderId: payload.parentFolderId,
+        folderName: payload.folderName,
+      }, {
+        cancelToken: cancelTokenSource.token,
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+
+    return [promise, cancelTokenSource];
+  }
+
+  /**
+   * Moves a specific folder to a new location
+   * @param payload
+   */
+  public async moveFolder(payload: MoveFolderPayload): Promise<MoveFolderResponse> {
+    return this.axios
+      .post(`${this.apiUrl}/api/storage/move/folder`, {
+        folderId: payload.folder.id,
+        destination: payload.destinationFolderId,
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+  }
+
+  /**
+   * Updates the metadata of a folder
+   * @param payload
+   */
+  public async updateFolder(payload: UpdateFolderMetadataPayload): Promise<void> {
+    await this.axios
+      .post(`${this.apiUrl}/api/storage/folder/${payload.folderId}/meta`, {
+        metadata: payload.changes
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+  }
+
+  /**
+   * Fetches & returns the contents of a specific folder
+   * @param folderId
+   */
+  public getFolderContent(folderId: number): [
+    Promise<{
+      folders: DriveFolderData[];
+      files: DriveFileData[]
+    }>,
+    CancelTokenSource
+  ] {
+    const cancelTokenSource = axios.CancelToken.source();
+    const promise = this.axios
+      .get<FetchFolderContentResponse>(`${this.apiUrl}/api/storage/v2/folder/${folderId}`, {
+        cancelToken: cancelTokenSource.token,
+        headers: this.headers()
+      })
+      .then(response => {
+        return {
+          folders: response.data.children.map(folder => ({ ...folder, isFolder: true })),
```

Comment:
This is something used by drive-web for its own purposes, the SDK should not add custom parameters like `isFolder`. Just return the content from the server in a convenient, platform-agnostic structure.